File: Examples/CSharp/Working-with-Pages/ExportOfHiddenVisioPagesToXPS.cs
using Aspose.Diagram;
using Aspose.Diagram.Saving;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Aspose.Diagram.Examples.CSharp.Working_with_Pages
{
    public class ExportOfHiddenVisioPagesToXPS
    {
        public static void Run()
        {
            try
            {
                // ExStart:ExportOfHiddenVisioPagesToXPS
                // The path to the documents directory.
                string dataDir = RunExamples.GetDataDir_Intro();

                // load an existing Visio
                Diagram diagram = new Diagram(dataDir + "Drawing1.vsdx");
                // get a particular page
                Page page = diagram.Pages.GetPage("Flow 2");
                // set Visio page visiblity
                page.PageSheet.PageProps.UIVisibility.Value = BOOL.False;

                // initialize PDF save options
                XPSSaveOptions options = new XPSSaveOptions();
                // set export option of hidden Visio pages
                options.ExportHiddenPage = false;

                // Save the Visio diagram
                diagram.Save(dataDir + "ExportOfHiddenVisioPagesToXPS_Out.xps", options);
                // ExEnd:ExportOfHiddenVisioPagesToXPS
            }
            catch (System.Exception ex)
            {
                System.Console.WriteLine(ex.Message + "\nThis example will only work if you apply a valid Aspose License. You can purchase full license or get 30 day temporary license from http:// Www.aspose.com/purchase/default.aspx.");
            }
        }
    }
}

File: Examples/VisualBasic/Working-with-Pages/ExportOfHiddenVisioPagesToXPS.vb
Imports Aspose.Diagram
Imports Aspose.Diagram.Saving
Imports System
Public Class ExportOfHiddenVisioPagesToXPS
    Public Shared Sub Run()
        Try
            ' ExStart:ExportOfHiddenVisioPagesToXPS
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_Intro()

            ' Load an existing Visio
            Dim diagram As New Diagram(dataDir & Convert.ToString("Drawing1.vsdx"))
            ' Get a particular page
            Dim page As Page = diagram.Pages.GetPage("Flow 2")
            ' Set Visio page visiblity
            page.PageSheet.PageProps.UIVisibility.Value = BOOL.True

            ' Initialize PDF save options
            Dim options As New XPSSaveOptions()
            ' Set export option of hidden Visio pages
            options.ExportHiddenPage = False

            ' Save the Visio diagram
            diagram.Save(dataDir & Convert.ToString("ExportOfHiddenVisioPagesToXPS_out.xps"), options)
            ' ExEnd:ExportOfHiddenVisioPagesToXPS
        Catch ex As System.Exception
            System.Console.WriteLine("This example will only work if you apply a valid Aspose License. You can purchase full license or get 30 day temporary license from http://www.aspose.com/purchase/default.aspx.")
        End Try
    End Sub
End Class
